File: C++/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P06_Cake.cpp
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

int main() {
	int a, b;
	cin >> a >> b;

	int total = a * b;
	for (; ; ) {
		string n;
		cin >> n;
		if (n == "STOP") {
			cout << total << " pieces are left." << endl;
			break;
		}
		else {
			int left = stoi(n);
			total -= left;
			if (total <= 0) {
				cout << "No more cake left! You need " << abs(total) << " pieces more." << endl;
				break;
			}
		}
	}

	return 0;
}

File: Python/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P06_Cake.py
width = int(input())
lenght = int(input())
count = width * lenght

while count > 0:
    line = input()

    if line == "STOP":
        break
    else:
        new_pieces = int(line)
        count = count - new_pieces

if count > 0:
    print(f"{count} pieces are left.")
else:
    print(f"No more cake left! You need {abs(count)} pieces more.")
